modules/light-clients/06-solomachine/store.go

```diff
@@ -18,5 +20,11 @@ func getClientState(store storetypes.KVStore, cdc codec.BinaryCodec) (*ClientSta
 	}
 
 	clientStateI := clienttypes.MustUnmarshalClientState(cdc, bz)
-	return clientStateI.(*ClientState), true
+	var clientState *ClientState
+	clientState, ok := clientStateI.(*ClientState)
+	if !ok {
+		panic(fmt.Errorf("cannot convert %T to %T", clientStateI, clientState))
```

Comment:
I guess if we wanted to avoid the pain we could return `nil, false`, but that's stretching maybe a bit too much the semantics of the function signature...

Reply:
I agree that `nil, false` would probably hide the issue. The other way would be to change the signature to return an error, but it seems this would become a much broader PR